runtime/lua/vim/lsp/handlers.lua

```diff
@@ -350,7 +350,9 @@ M['textDocument/signatureHelp'] = M.signature_help
 --see: https://microsoft.github.io/language-server-protocol/specifications/specification-current/#textDocument_documentHighlight
 M['textDocument/documentHighlight'] = function(_, result, ctx, _)
   if not result then return end
-  util.buf_highlight_references(ctx.bufnr, result, ctx.client_id)
+  local client_id = ctx.client_id
+  local client = vim.lsp.get_client_by_id(client_id)
+  util.buf_highlight_references(ctx.bufnr, result, client.offset_encoding)
```

Comment:
```suggestion
  util.buf_highlight_references(ctx.bufnr, result, client and client.offset_encoding)
```

I think client could be nil here because of a race between client shutting down and receiving the highlights. Or maybe we should just ignore the highlight in that case, not sure if they'd otherwise still get cleared if the client already shutdown

Reply:
We should just protect all handlers this way, I'll do this the quick fix for now. The issue with the above approach is if you are using clangd, it will apply to the wrong highlight range (running in utf-8 mode)